File: Hash Table/771. Jewels and Stones/jewels.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones)
    {
        map<char, bool> m;

        for(auto item: jewels)
        {
            m[item] = true;
        }
        int count = 0;

        for(auto item: stones)
        {
            if (m[item])
            {
                count++;
            }
        }
        return count;       
    }
};

File: 4. Hash Table/771. Jewels and Stones/jewels.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        
        charset = set()
        for char in jewels:
            charset.add(char)
        
        jewel = 0
        for char in stones:
            if char in charset:
                jewel += 1

        return jewel